File: classified/packages/plugin-plugin-manager/src/__tests__/e2e/registryIntegration.e2e.ts
// This test file is disabled as it was written for the old local registry service.
// The new API-based registry system is tested in src/test-registry.ts
// and through the API service tests in packages/api-service/

import { describe, it, expect, beforeEach, vi } from 'vitest';
import type { IAgentRuntime, Memory, UUID } from '@elizaos/core';
import { searchPluginAction, getPluginDetailsAction } from '../../actions/searchPluginAction';
import { clonePluginAction } from '../../actions/clonePluginAction';
import * as pluginRegistryService from '../../services/pluginRegistryService';

// Mock the entire pluginRegistryService
vi.mock('../../services/pluginRegistryService', () => ({
  searchPluginsByContent: vi.fn(),
  getPluginDetails: vi.fn(),
  clonePlugin: vi.fn(),
}));

describe('Registry Integration E2E Tests', () => {
  let runtime: IAgentRuntime;

  beforeEach(() => {
    runtime = {
      agentId: 'test-agent' as UUID,
      actions: [],
      registerAction: (action: any) => runtime.actions.push(action),
      getService: (name: any) => null,
    } as any;

    vi.clearAllMocks();
  });

  describe('searchPluginAction', () => {
    it('should call searchPluginsByContent and format results', async () => {
      const mockResults = [
        {
          name: '@elizaos/plugin-weather',
          description: 'A weather plugin',
          score: 0.9,
          tags: ['weather'],
        },
      ];
      (pluginRegistryService.searchPluginsByContent as any).mockResolvedValue(mockResults);

      const message = { content: { text: 'search for weather plugins' } } as Memory;
      const callback = vi.fn();

      await searchPluginAction.handler(runtime, message, undefined, undefined, callback);

      expect(pluginRegistryService.searchPluginsByContent).toHaveBeenCalledWith('weather plugins');
      expect(callback).toHaveBeenCalledWith(
        expect.objectContaining({
          text: expect.stringContaining('Found 1 plugin'),
        })
      );
    });
  });

  describe('getPluginDetailsAction', () => {
    it('should call getPluginDetails and format details', async () => {
      const mockDetails = {
        name: '@elizaos/plugin-weather',
        description: 'A weather plugin',
        version: '1.0.0',
        tags: ['weather'],
        features: ['current weather'],
      };
      (pluginRegistryService.getPluginDetails as any).mockResolvedValue(mockDetails);

      const message = { content: { text: 'get details for @elizaos/plugin-weather' } } as Memory;
      const callback = vi.fn();

      await getPluginDetailsAction.handler(runtime, message, undefined, undefined, callback);

      expect(pluginRegistryService.getPluginDetails).toHaveBeenCalledWith(
        '@elizaos/plugin-weather'
      );
      expect(callback).toHaveBeenCalledWith(
        expect.objectContaining({
          text: expect.stringContaining('**@elizaos/plugin-weather** Details'),
        })
      );
    });
  });

  describe('clonePluginAction', () => {
    it('should call clonePlugin and report success', async () => {
      const mockResult = {
        success: true,
        pluginName: '@elizaos/plugin-weather',
        localPath: '/cloned-plugins/plugin-weather',
      };
      (pluginRegistryService.clonePlugin as any).mockResolvedValue(mockResult);

      const message = { content: { text: 'clone plugin @elizaos/plugin-weather' } } as Memory;
      const callback = vi.fn();

      await clonePluginAction.handler(runtime, message, undefined, undefined, callback);

      expect(pluginRegistryService.clonePlugin).toHaveBeenCalledWith('@elizaos/plugin-weather');
      expect(callback).toHaveBeenCalledWith(
        expect.objectContaining({
          text: expect.stringContaining('Successfully cloned'),
        })
      );
    });
  });
});

File: cpp/generated_misc/src/registryIntegration.e2e.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/classified/packages/plugin-plugin-manager/src/__tests__/e2e/registryIntegration.e2e.h"

void Main(void)
{
    vi->mock(std::string("../../services/pluginRegistryService"), [=]() mutable
    {
        return (object{
            object::pair{std::string("searchPluginsByContent"), vi->fn()}, 
            object::pair{std::string("getPluginDetails"), vi->fn()}, 
            object::pair{std::string("clonePlugin"), vi->fn()}
        });
    }
    );
    describe(std::string("Registry Integration E2E Tests"), [=]() mutable
    {
        shared<std::shared_ptr<IAgentRuntime>> runtime;
        beforeEach([=]() mutable
        {
            runtime = as<any>(object{
                object::pair{std::string("agentId"), as<std::shared_ptr<UUID>>(std::string("test-agent"))}, 
                object::pair{std::string("actions"), array<any>()}, 
                object::pair{std::string("registerAction"), [=](auto action) mutable
                {
                    return runtime->actions->push(action);
                }
                }, 
                object::pair{std::string("getService"), [=](auto name) mutable
                {
                    return nullptr;
                }
                }
            });
            vi->clearAllMocks();
        }
        );
        describe(std::string("searchPluginAction"), [=]() mutable
        {
            it(std::string("should call searchPluginsByContent and format results"), [=]() mutable
            {
                auto mockResults = array<object>{ object{
                    object::pair{std::string("name"), std::string("@elizaos/plugin-weather")}, 
                    object::pair{std::string("description"), std::string("A weather plugin")}, 
                    object::pair{std::string("score"), 0.9}, 
                    object::pair{std::string("tags"), array<string>{ std::string("weather") }}
                } };
                (as<any>(pluginRegistryService->searchPluginsByContent))["mockResolvedValue"](mockResults);
                auto message = as<std::shared_ptr<Memory>>(object{
                    object::pair{std::string("content"), object{
                        object::pair{std::string("text"), std::string("search for weather plugins")}
                    }}
                });
                auto callback = vi->fn();
                std::async([=]() { searchPluginAction->handler(runtime, message, undefined, undefined, callback); });
                expect(pluginRegistryService->searchPluginsByContent)->toHaveBeenCalledWith(std::string("weather plugins"));
                expect(callback)->toHaveBeenCalledWith(expect->objectContaining(object{
                    object::pair{std::string("text"), expect->stringContaining(std::string("Found 1 plugin"))}
                }));
            }
            );
        }
        );
        describe(std::string("getPluginDetailsAction"), [=]() mutable
        {
            it(std::string("should call getPluginDetails and format details"), [=]() mutable
            {
                auto mockDetails = object{
                    object::pair{std::string("name"), std::string("@elizaos/plugin-weather")}, 
                    object::pair{std::string("description"), std::string("A weather plugin")}, 
                    object::pair{std::string("version"), std::string("1.0.0")}, 
                    object::pair{std::string("tags"), array<string>{ std::string("weather") }}, 
                    object::pair{std::string("features"), array<string>{ std::string("current weather") }}
                };
                (as<any>(pluginRegistryService->getPluginDetails))["mockResolvedValue"](mockDetails);
                auto message = as<std::shared_ptr<Memory>>(object{
                    object::pair{std::string("content"), object{
                        object::pair{std::string("text"), std::string("get details for @elizaos/plugin-weather")}
                    }}
                });
                auto callback = vi->fn();
                std::async([=]() { getPluginDetailsAction->handler(runtime, message, undefined, undefined, callback); });
                expect(pluginRegistryService->getPluginDetails)->toHaveBeenCalledWith(std::string("@elizaos/plugin-weather"));
                expect(callback)->toHaveBeenCalledWith(expect->objectContaining(object{
                    object::pair{std::string("text"), expect->stringContaining(std::string("**@elizaos/plugin-weather** Details"))}
                }));
            }
            );
        }
        );
        describe(std::string("clonePluginAction"), [=]() mutable
        {
            it(std::string("should call clonePlugin and report success"), [=]() mutable
            {
                auto mockResult = object{
                    object::pair{std::string("success"), true}, 
                    object::pair{std::string("pluginName"), std::string("@elizaos/plugin-weather")}, 
                    object::pair{std::string("localPath"), std::string("/cloned-plugins/plugin-weather")}
                };
                (as<any>(pluginRegistryService->clonePlugin))["mockResolvedValue"](mockResult);
                auto message = as<std::shared_ptr<Memory>>(object{
                    object::pair{std::string("content"), object{
                        object::pair{std::string("text"), std::string("clone plugin @elizaos/plugin-weather")}
                    }}
                });
                auto callback = vi->fn();
                std::async([=]() { clonePluginAction->handler(runtime, message, undefined, undefined, callback); });
                expect(pluginRegistryService->clonePlugin)->toHaveBeenCalledWith(std::string("@elizaos/plugin-weather"));
                expect(callback)->toHaveBeenCalledWith(expect->objectContaining(object{
                    object::pair{std::string("text"), expect->stringContaining(std::string("Successfully cloned"))}
                }));
            }
            );
        }
        );
    }
    );
}

MAIN
